XamlControlsGallery/ControlPages/GridViewPage.xaml.cs

```diff
@@ -201,357 +182,41 @@ private void SelectionModeComboBox_SelectionChanged(object sender, SelectionChan
-                ActualMaxItems -= 1;
-                MaxItems.Text = ActualMaxItems.ToString();
-                MaxItems.PlaceholderText = ActualMaxItems.ToString();
-
-                // Set property by accessing StyledGridIWG ItemsWrapGrid object, which is defined on load (via StyledGrid_InitWrapGrid fcn)
-                StyledGridIWG.MaximumRowsOrColumns = ActualMaxItems;
-            }
-
-            // If value is less than 1, reset textbox to 1 and display error message.
-            else
-            {
-                MaxItemsDec.IsEnabled = false;
-                ActualMaxItems = 1;
-                MaxItems.Text = "1";
-                MaxItems.PlaceholderText = "1";
-            }
-        }
-
-        private void StyledGrid_ChangeMaxItems(object sender, TextChangedEventArgs e)
-        {
-            try
-            {
-                // Make sure that value entered is a number
-                int newVal = Convert.ToInt32(MaxItems.Text.ToString());
-
-                // Make sure number is not less than 1
-                if (Convert.ToInt32(MaxItems.Text.ToString()) > 1)
-                {
-                    MaxItemsDec.IsEnabled = true;
-                    MaxItems.PlaceholderText = MaxItems.Text.ToString();
-                    ActualMaxItems = Convert.ToInt32(MaxItems.Text);
-
-                    
-                    // Set property by accessing StyledGridIWG ItemsWrapGrid object, which is defined on load (via StyledGrid_InitWrapGrid fcn)
-                    StyledGridIWG.MaximumRowsOrColumns = ActualMaxItems;
-                }
-
-                // If number is less than 1, reset textbox to 1 and display error message.
-                else
-                {
-                    MaxItemsDec.IsEnabled = false;
-                    ActualMaxItems = 1;
-                    MaxItems.PlaceholderText = "1";
-                    MaxItems.Text = "1";
-                    StyledGridIWG.MaximumRowsOrColumns = ActualMaxItems;
-                }
-
-            }
-            
-            // If value not a number, reset text box to 1.
-            catch (FormatException)
-            {
-                if (MaxItems.Text == "")
-                {
-                    StyledGridIWG.MaximumRowsOrColumns = 1;
-                    MaxItemsDec.IsEnabled = false;
-                }
-                MaxItems.Text = "";
-                MaxItems.PlaceholderText = "1";
-                ActualMaxItems = 1;
-            }
-            
-        }
-
-        private void StyledGrid_InitWrapGrid(object sender, RoutedEventArgs e)
-        {
-            // Update ItemsWrapGrid object created on page load by assigning it to StyledGrid's ItemWrapGrid
-            StyledGridIWG = sender as ItemsWrapGrid;
-
-            // Now we can change StyledGrid's MaximumRowsorColumns property within its ItemsPanel>ItemsPanelTemplate>ItemsWrapGrid.
-            StyledGridIWG.MaximumRowsOrColumns = ActualMaxItems;
```

Comment:
This is a ton of deleted code. Double checking that we don't need it. Why don't we need it?

Reply:
Reading a bit deeper: is it because a lot of the deleted code is just checking that the value that the user inputs is valid?

Reply:
Yes, a lot of the deleted code was needed to parse numbers, ignore text and keep the number in a certain range. With the NumberBox control, all of that is not needed anymore.